### PR TITLE
feat: add check_run collector in github graphql

### LINKED ISSUE #3704: [Feature][github_graphql] collect workflow jobs data by graphql
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-devlake/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

collect workflow jobs data by graphql.
Now GitHub restful plugin requests workflow jobs data from `run/workflowRunId/jobs`. It need to request jobs in each workflow run.
So in devlake jobsCollector takes 30000+ requests.

### Use case

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by likyh**
# Summary

CheckSuite.CheckRuns have some columns which are similar to WorkflowJobs.

So collect `CheckRuns` on each CheckSuite to replace WorkflowJobCollector.

Now query 30 CheckSuite in one graphql request. This new collector costs about 40min to collect Devlake's runs & jobs on my computer, and its speed is about 1000/min (CheckSuite) or 2 requests per second.

### Does this close any open issues?
Closes #3704

### Screenshots

The result in `_tool_github_jobs`:

| created\_at | updated\_at | \_raw\_data\_params | \_raw\_data\_table | \_raw\_data\_id | \_raw\_data\_remark | connection\_id | repo\_id | id | run\_id | run\_url | node\_id | head\_sha | url | html\_url | status | conclusion | started\_at | completed\_at | name | steps | check\_run\_url | labels | runner\_id | runner\_name | runner\_group\_id | type |
| :--- | :--- | :--- | :--- | :--- | :--- | :--- | :--- | :--- | :--- | :--- | :--- | :--- | :--- | :--- | :--- | :--- | :--- | :--- | :--- | :--- | :--- | :--- | :--- | :--- | :--- | :--- |
| 2022-11-10 06:11:16.391154 +00:00 | 2022-11-10 06:11:16.391154 +00:00 | {"ConnectionId":1,"Owner":"apache","Repo":"incubator-devlake"} | \_raw\_github\_graphql\_check\_runs | 380 |  | 1 | 384111310 | 9267762757 | 3384137147 |  | CR\_kwDOFuUSzs8AAAACKGbWRQ |  |  | https://github.com/apache/incubator-devlake/actions/runs/3384137147/jobs/5620754276 | COMPLETED | SUCCESS | 2022-11-03 08:40:38.000000 +00:00 | 2022-11-03 08:40:57.000000 +00:00 | commit-msg | ......Look at below |  | NULL | 0 |  | 0 |  |
| 2022-11-10 06:11:16.391154 +00:00 | 2022-11-10 06:11:16.391154 +00:00 | {"ConnectionId":1,"Owner":"apache","Repo":"incubator-devlake"} | \_raw\_github\_graphql\_check\_runs | 380 |  | 1 | 384111310 | 9266543183 | 3383690109 |  | CR\_kwDOFuUSzs8AAAACKFQ6Tw |  |  | https://github.com/apache/incubator-devlake/actions/runs/3383690109/jobs/5619825185 | COMPLETED | SUCCESS | 2022-11-03 07:14:06.000000 +00:00 | 2022-11-03 07:15:51.000000 +00:00 | lint for yamls | ......Look at below |  | NULL | 0 |  | 0 |  |

Steps.1:

```json
[{"name": "Set up job", "number": 1, "status": "COMPLETED", "conclusion": "SUCCESS", "started_at": "2022-11-03T08:40:38Z", "completed_at": "2022-11-03T08:40:41Z", "seconds_to_completion": 3}, {"name": "Run actions/checkout@v2", "number": 2, "status": "COMPLETED", "conclusion": "SUCCESS", "started_at": "2022-11-03T08:40:41Z", "completed_at": "2022-11-03T08:40:52Z", "seconds_to_completion": 11}, {"name": "Cache lint-commit-message", "number": 3, "status": "COMPLETED", "conclusion": "SUCCESS", "started_at": "2022-11-03T08:40:52Z", "completed_at": "2022-11-03T08:40:55Z", "seconds_to_completion": 3}, {"name": "Lint commit message", "number": 4, "status": "COMPLETED", "conclusion": "SUCCESS", "started_at": "2022-11-03T08:40:56Z", "completed_at": "2022-11-03T08:40:56Z", "seconds_to_completion": 0}, {"name": "Post Cache lint-commit-message", "number": 7, "status": "COMPLETED", "conclusion": "SUCCESS", "started_at": "2022-11-03T08:40:56Z", "completed_at": "2022-11-03T08:40:56Z", "seconds_to_completion": 0}, {"name": "Post Run actions/checkout@v2", "number": 8, "status": "COMPLETED", "conclusion": "SUCCESS", "started_at": "2022-11-03T08:40:56Z", "completed_at": "2022-11-03T08:40:56Z", "seconds_to_completion": 0}, {"name": "Complete job", "number": 9, "status": "COMPLETED", "conclusion": "SUCCESS", "started_at": "2022-11-03T08:40:56Z", "completed_at": "2022-11-03T08:40:56Z", "seconds_to_completion": 0}]
```

Steps.2:
```json
[{"name": "Set up job", "number": 1, "status": "COMPLETED", "conclusion": "SUCCESS", "started_at": "2022-11-03T07:14:05Z", "completed_at": "2022-11-03T07:14:06Z", "seconds_to_completion": 1}, {"name": "Pull ghcr.io/github/super-linter:slim-v4.9.7", "number": 2, "status": "COMPLETED", "conclusion": "SUCCESS", "started_at": "2022-11-03T07:14:06Z", "completed_at": "2022-11-03T07:15:17Z", "seconds_to_completion": 71}, {"name": "Run actions/checkout@v3", "number": 3, "status": "COMPLETED", "conclusion": "SUCCESS", "started_at": "2022-11-03T07:15:17Z", "completed_at": "2022-11-03T07:15:18Z", "seconds_to_completion": 1}, {"name": "Run github/super-linter/slim@v4", "number": 4, "status": "COMPLETED", "conclusion": "SUCCESS", "started_at": "2022-11-03T07:15:18Z", "completed_at": "2022-11-03T07:15:50Z", "seconds_to_completion": 32}, {"name": "Post Run actions/checkout@v3", "number": 8, "status": "COMPLETED", "conclusion": "SUCCESS", "started_at": "2022-11-03T07:15:50Z", "completed_at": "2022-11-03T07:15:50Z", "seconds_to_completion": 0}, {"name": "Complete job", "number": 9, "status": "COMPLETED", "conclusion": "SUCCESS", "started_at": "2022-11-03T07:15:50Z", "completed_at": "2022-11-03T07:15:50Z", "seconds_to_completion": 0}]
```

### CODE REVIEW

File: plugins/github_graphql/tasks/check_run_collector.go [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+				Owner:        data.Options.Owner,
+				Repo:         data.Options.Repo,
+			},
+			Table: RAW_CHECK_RUNS_TABLE,
+		},
+		Input:         iterator,
+		InputStep:     30,
+		GraphqlClient: data.GraphqlClient,
+		BuildQuery: func(reqData *helper.GraphqlRequestData) (interface{}, map[string]interface{}, error) {
+			accounts := reqData.Input.([]interface{})
```
Comment (warren830) [inline plugins/github_graphql/tasks/check_run_collector.go:126]: should this be workflowRun?

File: plugins/github_graphql/tasks/check_run_collector.go [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+			query := &GraphqlQueryCheckRunWrapper{}
+			checkSuiteIds := []map[string]interface{}{}
+			for _, iAccount := range accounts {
```
Comment (warren830) [inline plugins/github_graphql/tasks/check_run_collector.go:129]: same as above

File: plugins/github_graphql/tasks/check_run_collector.go [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+						NodeID:       checkRun.Id,
+						HTMLURL:      checkRun.DetailsUrl,
+						Status:       checkRun.Status,
+						Conclusion:   checkRun.Conclusion,
+						StartedAt:    checkRun.StartedAt,
+						CompletedAt:  checkRun.CompletedAt,
+						Name:         checkRun.Name,
+						Steps:        paramsBytes,
+						// these columns can not fill by graphql
+						//RunID:         ``,  // use _tool_github_runs
```
Comment (warren830) [inline plugins/github_graphql/tasks/check_run_collector.go:169]: I think we also need to set RunId as this will be used to connect to pipelines

**Review by warren830 — COMMENTED** [review]
On line 99 in plugins/github_graphql/tasks/check_run_collector.go
we can also select workflowRunId, and set this value to job.RunId

File: plugins/github_graphql/plugin_main.go [hunk 4]
```diff
@@ -75,6 +80,7 @@ func (plugin GithubGraphql) SubTaskMetas() []core.SubTaskMeta {
 
 		tasks.CollectAccountMeta,
```
Comment (warren830) [inline plugins/github_graphql/plugin_main.go:82]: please also delete these blank lines

File: plugins/github_graphql/tasks/check_run_collector.go [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+						//HeadSha:       ``,  // use _tool_github_runs
+						//RunURL:        ``,
+						//CheckRunURL:   ``,
+						//Labels:        ``, // not on use
+						//RunnerID:      ``, // not on use
+						//RunnerName:    ``, // not on use
+						//RunnerGroupID: ``, // not on use
```
Comment (warren830) [inline plugins/github_graphql/tasks/check_run_collector.go:176]: please delete these unused code

File: plugins/github_graphql/tasks/check_run_collector.go [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+			Ctx: taskCtx,
+			Params: githubTasks.GithubApiParams{
+				ConnectionId: data.Options.ConnectionId,
+				Owner:        data.Options.Owner,
+				Repo:         data.Options.Repo,
+			},
+			Table: RAW_CHECK_RUNS_TABLE,
+		},
+		Input:         iterator,
+		InputStep:     30,
```
Comment (warren830) [inline plugins/github_graphql/tasks/check_run_collector.go:123]: how about 60?

File: plugins/github_graphql/tasks/check_run_collector.go (hunk 1 above)
Comment (likyh) [reply plugins/github_graphql/tasks/check_run_collector.go:126]: fixed

File: plugins/github_graphql/tasks/check_run_collector.go (hunk 2 above)
Comment (likyh) [reply plugins/github_graphql/tasks/check_run_collector.go:129]: fixed

File: plugins/github_graphql/tasks/check_run_collector.go (hunk 3 above)
Comment (likyh) [reply plugins/github_graphql/tasks/check_run_collector.go:169]: fixed

File: plugins/github_graphql/plugin_main.go (hunk 4 above)
Comment (likyh) [reply plugins/github_graphql/plugin_main.go:82]: add some comments here

File: plugins/github_graphql/tasks/check_run_collector.go (hunk 5 above)
Comment (likyh) [reply plugins/github_graphql/tasks/check_run_collector.go:176]: that is to tell other developers where we can fill runner info if they want.

File: plugins/github_graphql/tasks/check_run_collector.go (hunk 6 above)
Comment (likyh) [reply plugins/github_graphql/tasks/check_run_collector.go:123]: ok

**Review by warren830 — APPROVED** [review]
LGTM

**Pull request merged**